pytorch_lightning/plugins/environments/lsf_environment.py

```diff
@@ -51,112 +54,152 @@ def __init__(self):
             )
         self._main_address = self._get_main_address()
         self._main_port = self._get_main_port()
-        log.debug(f"MASTER_ADDR: {self._main_address}")
-        log.debug(f"MASTER_PORT: {self._main_port}")
-
-    @property
-    def creates_processes_externally(self) -> bool:
-        return True
-
-    @property
-    def main_address(self) -> str:
-        """The main address is read from a list of hosts contained in the environment variable `LSB_HOSTS`."""
-        return self._main_address
-
-    @property
-    def main_port(self) -> int:
-        """The main port gets calculated from the LSF job ID."""
-        return self._main_port
+        self._local_rank = self._get_local_rank()
+        self._global_rank = self._get_global_rank()
+        self._world_size = self._get_world_size()
+        self._node_rank = self._get_node_rank()
+
+        # set environment variables needed for initializing torch distributed process group
+        os.environ["MASTER_ADDR"] = str(self._main_address)
+        log.debug(f"MASTER_ADDR: {os.environ['MASTER_ADDR']}")
+        os.environ["MASTER_PORT"] = str(self._main_port)
+        log.debug(f"MASTER_PORT: {os.environ['MASTER_PORT']}")
+
+        tmp = ("main_address", "main_port", "world_size", "local_rank", "node_rank", "global_rank")
+        self._rep = ",".join("{}={}".format(s, getattr(self, "_" + s)) for s in tmp)
+
+    def _read_hosts(self):
+        var = "LSB_DJOB_RANKFILE"
+        try:
+            rankfile = os.environ[var]
+        except KeyError:
+            raise ValueError("Could not find environment variable LSB_DJOB_RANKFILE")
+        if not rankfile:
+            raise ValueError("Environment variable LSB_DJOB_RANKFILE is empty")
+        with open(rankfile) as f:
+            ret = [line.strip() for line in f]
+        return ret
+
+    def _get_main_address(self):
+        """A helper for getting the master address."""
+        hosts = self._read_hosts()
+        return hosts[1]
```

Comment:
Should we validate that `batch` is hosts[0] ?

Reply:
No. In fact I am going to remove this check. I am not certain that `batch` will always be used to launch nodes. I removed this check.